shared/test/test_redis_property_bag.rb

```diff
@@ -133,35 +140,57 @@ def test_deferred_expiration
     assert_equal({'foo1' => 'value1', 'foo2' => 'value2', 'foo3' => 'value3'}, bag.to_hash)
 
     # Jump to just before expiration
-    sleep test_delay_seconds - 0.1
+    time_travel test_delay_seconds - 0.1
     assert_equal({'foo1' => 'value1', 'foo2' => 'value2', 'foo3' => 'value3'}, bag.to_hash)
 
     # Perform a write to reset the expire time
     bag.set('foo1', 'value4')
     assert_equal({'foo1' => 'value4', 'foo2' => 'value2', 'foo3' => 'value3'}, bag.to_hash)
 
     # Jump forward in time to original expiration
-    sleep 0.1
+    time_travel 0.1
     assert_equal({'foo1' => 'value4', 'foo2' => 'value2', 'foo3' => 'value3'}, bag.to_hash)
 
     # Jump to just before new expiration
-    sleep test_delay_seconds - 0.2
+    time_travel test_delay_seconds - 0.2
     assert_equal({'foo1' => 'value4', 'foo2' => 'value2', 'foo3' => 'value3'}, bag.to_hash)
 
     # Perform a delete to reset the expire time
     bag.delete(['foo2'])
     assert_equal({'foo1' => 'value4', 'foo3' => 'value3'}, bag.to_hash)
 
     # Jump forward in time to original expiration
-    sleep 0.1
+    time_travel 0.1
     assert_equal({'foo1' => 'value4', 'foo3' => 'value3'}, bag.to_hash)
 
     # Jump to just before new expiration
-    sleep test_delay_seconds - 0.2
+    time_travel test_delay_seconds - 0.2
     assert_equal({'foo1' => 'value4', 'foo3' => 'value3'}, bag.to_hash)
 
     # Jump forward in time to new expiration
-    sleep 0.2 # 0.1s margin of error for real redis
+    time_travel 0.2 # 0.1s margin of error for real redis
     assert_equal({}, bag.to_hash)
+  ensure
+    resume_time
+  end
+
+  private
+
+  # Set of helpers for using Timecop if we're using fakeredis, or passing
+  # real time if we're using real redis.
+  def stop_time
+    Timecop.freeze unless use_real_redis?
+  end
+
+  def resume_time
+    Timecop.return unless use_real_redis?
+  end
+
+  def time_travel(seconds)
+    if use_real_redis?
+      sleep seconds
+    else
+      Timecop.travel seconds
+    end
```

Comment:
DRY. Any chance we could move this into a time utility file? Presumably with a parameter `use_timecop`?